### ISSUE TITLE
Wired: DHCP: not gaining an address

### ISSUE THREAD

**Issue by grahamperrin** [ghostbsd/networkmgr#56]
https://www.freshports.org/net-mgmt/networkmgr/ networkmgr `5.0` on FreeBSD 13.0-RELEASE with `latest` packages. 

With the package installed: 

* after starting the system, an address is not gained until networkmgr is used to manually connect. 

After deleting the package and restarting the system: 

* an address is gained automatically. 

https://bsd-hardware.info/?probe=ce8d45af17 `RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller` driven by [re(4)](https://man.freebsd.org/re(4)). 

### Comparisons

https://forums.ghostbsd.org/viewtopic.php?p=9437#p9437

Also possibly [DHCP no longer working for em(4)](https://lists.freebsd.org/pipermail/freebsd-current/2021-April/079561.html) – `5.0` on FreeBSD `14.0-CURRENT`; I'll proceed to test without networkgr …

**Comment by grahamperrin** [comment]
## em(4)

### 82579LM Gigabit Network Connection (Lewisville)

> Also possibly [DHCP no longer working for em(4)](https://lists.freebsd.org/pipermail/freebsd-current/2021-April/079561.html) – `5.0` on FreeBSD `14.0-CURRENT`; I'll proceed to test without networkgr …

Confirmed. HP EliteBook 8570p. A few days ago, without understanding the cause(s) of the issue, I began working around by configuring [rc.conf(5)](https://man.freebsd.org/rc.conf(5)) to use static addressing with the DHCP service of my router at home. 

Today I reconfigured to use DHCP, after which `service netif restart em0` did not gain a network address. Worked around by using networkmgr to manually connect. 

I deleted the package, restarted, gained an address automatically. 

### 82540EM Gigabit Ethernet Controller

Whilst I do not have GhostBSD (with OpenRC) on real hardware at the time of writing, https://bsd-hardware.info/?probe=4bfec050c1 (2021-05-01, limited lifetime) is for VirtualBox, with the default `82540EM Gigabit Ethernet Controller` virtualised. 

Here, as far as I can tell, no problem.

**Comment by ericbsd** [comment]
@grahamperrin just to be clear those issues are is a booth on FreeBSD?

**Comment by grahamperrin** [comment]
Correct: 

* `RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller` driven by re(4) is FreeBSD `13.0-RELEASE` on an Ergo Vista 631 notebook (sometimes known as Asus T12Ah or Pegatron T12Ah), circa 2008
* `82579LM Gigabit Network Connection (Lewisville)` driven by em(0) is FreeBSD `14.0-CURRENT` on an HP EliteBook 8570p notebook, circa 2013. 

Will it help if I make the issue reproducible, on either computerm with GhostBSD `13-RELEASE`? I can try to install it to a spare USB flash drive (16 GB Kingston DataTraveler G3, I know it will be slow, but I don't mind, for testing).

**Comment by ericbsd** [comment]
> Will it helps if I make the issue reproducible, on either computer, with GhostBSD `13-RELEASE`? I can try to install it to a spare USB flash drive (16 GB Kingston DataTraveler G3, I know it will be slow, but I don't mind, for testing).

not really since they do not use the same thing to start the DHCP.

**Comment by ericbsd** [comment]
I was able to reproduce for GhostBSD and fix it. The only thing I do not use FreeBSD on desktop since I eat my own dog food.

**Comment by ericbsd** [comment]
I will test my fix for GhostBSD on my laptop.

**Comment by ericbsd** [comment]
this will fix the issue for GhostBSD https://github.com/ghostbsd/networkmgr/releases/tag/5.4

**Comment by ericbsd** [comment]
This morning's update should cover the problem for GhostBSD.

**Comment by ericbsd** [comment]
https://forums.ghostbsd.org/viewtopic.php?p=9454#p9454

**Comment by ericbsd** [comment]
For FreeBSD, I will have to install it in one of my hardware, when I have time.

**Comment by ericbsd** [comment]
I think I have found a fix for FreeBSD I will need help for testing when it is ready.

**Comment by grahamperrin** [comment]
Thanks, I'll help in that.

**Comment by ericbsd** [comment]
This PR https://github.com/ghostbsd/networkmgr/pull/65 should fix the problem for FreeBSD.

**Comment by ericbsd** [comment]
This should do it for FreeBSD https://www.freshports.org/net-mgmt/networkmgr/

**Comment by grahamperrin** [comment]
networkmgr-5.5 installed, wired network (em0) OK with a DHCP server following restart of FreeBSD 14.0-CURRENT. 

Thank you, would you like to close this issue?

**Comment by ericbsd** [comment]
Yeah, it looks fix to me, and it has improved the GhostBSD boot process with networking.

Thanks for testing that for me.
